txmongo/_gridfs/__init__.py

```diff
@@ -177,6 +177,19 @@ def ok(cursor):
         return self.__files.find(query, filter=filter.sort(myorder), limit=1, skip=skip)\
             .addCallback(ok)
 
+    @defer.inlineCallbacks
+    def count(self, filename=None):
+        """Count the number of versions of a given file.
+        Returns an integer number of versions of the file in GridFS whose filename matches
+        `filename`, or raises NoFile if the file doesn't exist.
+        :Parameters:
+          - `filename`: ``"filename"`` of the file to get version count of
+        """
+        cursor = yield self.__files.find({"filename": filename})
```

Comment:
It would be more efficient to call `self.__files.count({"filename": filename})` here. Not just because this way we won't move any file metadata across the TCP connection, but what is more important, MongoDB will be able to give us an answer only by querying index and not touching actual collection data at all.